### ISSUE TITLE
extensions should ignore imports from packages with invalid names

### ISSUE THREAD

**Issue by novemberborn** [benmosher/eslint-plugin-import#341]
Similar to #340, the extensions rule should ignore imports from packages with invalid names. E.g. [`files:../../content/**/*.md`](https://github.com/novemberborn/novemberborn.net/blob/949ae651597a017d34a6c93da094e9b8fbd0bf60/src/lib/content.js#L8) is not a file path, and not a valid package name.

I suppose this might have to deal with Windows file paths though.

**Comment by jfmengels** [comment]
Hi @novemberborn :)

Same as #340, do you have an example of a case where this is problematic for you? I'm not sure what kind of problem you're facing here.

**Comment by novemberborn** [comment]
Sure. I maintain two Babel plugins which rewrite import statements: [`import-glob`](https://github.com/novemberborn/babel-plugin-import-glob) and [`files`](https://github.com/novemberborn/babel-plugin-files). An example of the latter [is here](https://github.com/novemberborn/novemberborn.net/blob/949ae651597a017d34a6c93da094e9b8fbd0bf60/src/lib/content.js#L8): 

```js
import files from 'files:../../content/**/*.md'
```

The import isn't a file path (no leading / or disk drive, no leading period), and neither is `files:..` a valid package name. Since the `extensions` rule only deals with file extensions, and this isn't a valid import (because it uses a custom resolver), I don't think the rule should apply.

I suppose this might be solved by writing a resolver for these Babel plugins (though I haven't checked whether this rule follows resolvers). It might be useful for this rule not to result in errors even if there is no custom resolver.

**Comment by ljharb** [comment]
The proper solution here is indeed to write a resolver.
